dali/util/device_guard.h

```diff
@@ -30,13 +30,17 @@ class DeviceGuard {
     CUDA_CALL(cudaGetDevice(&original_device_));
     CUDA_CALL(cudaSetDevice(new_device));
   }
+
+
   ~DeviceGuard() noexcept(false) {
     CUDA_CALL(cudaSetDevice(original_device_));
```

Comment:
Throwing from a destructor is evil. How on Earth is it ever going to fail? Even more importantly - can we do anything about it? If we can't - why not just terminate the program here?

Reply:
Changed to abort